### DISCUSSION TITLE
Prisma and Supabase when `npx prisma migrate dev` not working

### DISCUSSION THREAD

**Discussion by enyelsequeira** [prisma/prisma#17058] [Q&A]
Hello all,

I recently started working with Prisma and Supabase, however I came around this issue and I am not sure whether this is part of `prisma` or `supabase` seems like an issue with Prisma. I am able to push my schema and works correctly however when running the command `prisma migrate dev` does not work and I get stuck without any feedback, I have found other faced similar issue wiht this command and tried other alternatives to debug but nothing 

![image](https://user-images.githubusercontent.com/50872387/209978953-29e57ac5-5016-4570-b2cb-038fdfc76bd8.png)
![image](https://user-images.githubusercontent.com/50872387/209979011-40da6a12-9205-4461-abc0-f727ece5eb1e.png)

env 
# Prisma env
DATABASE_URL=postgresql://postgres:postgres@localhost:5432/postgres

and schema 
![image](https://user-images.githubusercontent.com/50872387/209979148-d69a99f5-3326-4fc1-bdaf-071f6adc5b75.png)

**Comment by nurul3101** [comment]
Hello @enyelsequeira 👋 

Is this happening on the latest version of Prisma i.e. `4.8.0`?
If I understand correctly, you are able to use `npx prisma db push` as expected, but only `npx prisma migrate dev` command gives no output, correct?

**Reply by enyelsequeira** [reply]
hello @nurul3101 Actually took a while, but it seems like you have to do something like this

`  shadowDatabaseUrl = env("SHADOW_DATABASE_URL")`

**Reply by nurul3101** [reply]
Is your shadowDatabaseUrl different from the actual database URL?

**Reply by enyelsequeira** [reply]
on local it would look like this, and similar if it was using `DATABASE_URL="postgresql://postgres:postgres@localhost:54322/postgres"
SHADOW_DATABASE_URL="postgresql://postgres:postgres@localhost:54322/postgres/postgres_shadow"
`

**Reply by clarencejychan** [reply]
@enyelsequeira  were you able to resolve this? 

I ended up getting the first migration to work but any subsequent migration ends up deleting the prisma migration table under the `public` schema. Then I'll get the following error from running. 

```
npx prisma migrate dev --name add_onboarded

Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "postgres", schema "public" at "localhost:54322"

Error: P3005

The database schema is not empty. Read more about how to baseline an existing production database: https://pris.ly/d/migrate-baseline
```

This is on prisma 4.8.0.

**Reply by Twillied** [reply]
I'm experiencing the same issue - any suggestions?

**Reply by enyelsequeira** [reply]
Hey @clarencejychan and @Twillied yeah, I was able to fix that after creating a shadow_db

**Reply by Twillied** [reply]
@enyelsequeira - thanks - migrate dev now working as expected!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
